pkg/cmd/pr/merge/merge.go

```diff
@@ -168,224 +168,326 @@ func NewCmdMerge(f *cmdutil.Factory, runF func(*MergeOptions) error) *cobra.Comm
 
-			baseRemote, err := remotes.FindByRepo(baseRepo.RepoOwner(), baseRepo.RepoName())
+			baseRemote, err := remotes.FindByRepo(m.baseRepo.RepoOwner(), m.baseRepo.RepoName())
 			if err != nil {
 				return err
 			}
 
-			if git.HasLocalBranch(branchToSwitchTo) {
-				if err := git.CheckoutBranch(branchToSwitchTo); err != nil {
+			if git.HasLocalBranch(m.currentBranch) {
+				if err := git.CheckoutBranch(m.currentBranch); err != nil {
 					return err
 				}
 			} else {
-				if err := git.CheckoutNewBranch(baseRemote.Name, branchToSwitchTo); err != nil {
+				if err := git.CheckoutNewBranch(baseRemote.Name, m.currentBranch); err != nil {
 					return err
 				}
 			}
 
-			if err := git.Pull(baseRemote.Name, branchToSwitchTo); err != nil {
-				fmt.Fprintf(opts.IO.ErrOut, "%s warning: not possible to fast-forward to: %q\n", cs.WarningIcon(), branchToSwitchTo)
+			if err := git.Pull(baseRemote.Name, m.currentBranch); err != nil {
+				m.log(fmt.Sprintf("%s warning: not possible to fast-forward to: %q\n", m.cs.WarningIcon(), m.currentBranch))
 			}
 		}
 
-		if err := git.DeleteLocalBranch(pr.HeadRefName); err != nil {
-			err = fmt.Errorf("failed to delete local branch %s: %w", cs.Cyan(pr.HeadRefName), err)
+		if err := git.DeleteLocalBranch(m.pr.HeadRefName); err != nil {
+			err = fmt.Errorf("failed to delete local branch %s: %w", m.cs.Cyan(m.pr.HeadRefName), err)
 			return err
 		}
+	}
 
-		if branchToSwitchTo != "" {
-			branchSwitchString = fmt.Sprintf(" and switched to branch %s", cs.Cyan(branchToSwitchTo))
-		}
+	return nil
+}
+
+// Delete the remote branch if requested and if allowed.
+func (m *MergeContext) deleteRemoteBranch() error {
+	// the user was already asked if they want to delete the branch if they didn't provide the flag
+	if !m.deleteBranch || m.crossRepoPR || m.autoMerge {
+		return nil
 	}
-	if !isPRAlreadyMerged {
-		err = api.BranchDeleteRemote(apiClient, baseRepo, pr.HeadRefName)
+
+	if !m.merged {
+		err := api.BranchDeleteRemote(m.apiClient, m.baseRepo, m.pr.HeadRefName)
 		var httpErr api.HTTPError
 		// The ref might have already been deleted by GitHub
 		if err != nil && (!errors.As(err, &httpErr) || httpErr.StatusCode != 422) {
-			err = fmt.Errorf("failed to delete remote branch %s: %w", cs.Cyan(pr.HeadRefName), err)
-			return err
+			return fmt.Errorf("failed to delete remote branch %s: %w", m.cs.Cyan(m.pr.HeadRefName), err)
 		}
 	}
 
-	if isTerminal {
-		fmt.Fprintf(opts.IO.ErrOut, "%s Deleted branch %s%s\n", cs.SuccessIconWithColor(cs.Red), cs.Cyan(pr.HeadRefName), branchSwitchString)
+	if m.isTerminal {
+		branch := ""
+		if m.currentBranch != "" {
+			branch = fmt.Sprintf(" and switched to branch %s", m.cs.Cyan(m.currentBranch))
+		}
+		m.log(fmt.Sprintf("%s Deleted branch %s%s\n", m.cs.SuccessIconWithColor(m.cs.Red), m.cs.Cyan(m.pr.HeadRefName), branch))
+	}
+
+	return nil
+}
+
+// Log operations and results to the error stream.
+func (m *MergeContext) log(s string) {
+	fmt.Fprintf(m.opts.IO.ErrOut, s)
```

Comment:
I would like to modify this to
```suggestion
  if m.isTerminal {
	fmt.Fprintf(m.opts.IO.ErrOut, s)
  }
```
if we should only log when it is a terminal